app/components/c-attribute-grid.vue

```diff
@@ -0,0 +1,176 @@
+<template>
+  <div :class="b(modifiers)">
+    <div ref="table" :class="b('table', { open: isOpen })" :style="{ 'max-height': maxHeight }">
+      <c-attribute-grid-items :items="attributes"/>
+    </div>
+    <div :class="b('toggle', { open: isOpen })" role="button" @click="toggle()">
+      <e-icon :class="b('arrow')" inline icon="i-arrow-down"/>
+    </div>
+  </div>
+</template>
+
+<script>
+  import cAttributeGridItems from '@/components/c-attribute-grid-items';
+
+  export default {
+    name: 'c-attribute-grid',
+    components: {
+      cAttributeGridItems,
+    },
+    // mixins: [],
+
+    props: {
+      /**
+       * Defines the items displayed in the attribute grid.
+       */
+      attributes: {
+        type: Array,
+        required: true,
+      },
+
+      /**
+       * Defines whether the grid should shrink on mobile or not.
+       */
+      shrinkOnMobile: {
+        type: Boolean,
+        default: false,
+      },
+    },
+
+    data() {
+      return {
+        isOpen: true,
+        openHeight: 0,
+        closedHeight: 0,
+      };
+    },
+
+    computed: {
+      modifiers() {
+        return {
+          shrinkOnMobile: this.shrinkOnMobile,
+          dontShrinkOnMobile: !this.shrinkOnMobile
+        };
+      },
+      maxHeight() {
+        if (!this.shrinkOnMobile) {
+          return 'inherit';
+        }
+
+        if (this.isOpen) {
+          return this.openHeight;
+        }
+
+        return this.closedHeight;
+      },
+    },
+    // watch: {},
+
+    // beforeCreate() {},
+    // created() {},
+    // beforeMount() {},
+
+    mounted() {
+      this.recalculateOpenHeight();
+      window.addEventListener('resize', this.recalculateOpenHeight);
+
+      this.recalculateClosedHeight();
+      window.addEventListener('resize', this.recalculateClosedHeight);
+
+      this.isOpen = false;
+    },
+    // beforeUpdate() {},
+    // updated() {},
+    // activated() {},
+    // deactivated() {},
+    // beforeDestroy() {},
+    // destroyed() {},
+
+    methods: {
+      toggle() {
+        this.isOpen = !this.isOpen;
+      },
+      recalculateOpenHeight() {
+        let openHeight = this.$el.querySelector('dl').clientHeight;
```

Comment:
- [ ] Use $ref selector instead of query selector `<dl ref="list">` - `this.$refs.list`

Reply:
Does not work here, because the dl is in the child component